enhancements/testing/improved-platform-tests.md

```diff
@@ -0,0 +1,438 @@
+issues found, which also affected the increased delays.
+
+By separating OpenShift tests from the ones coming from kubernetes we will:
+- shorten the time when we start running most up-to-date tests from kubernetes;
+- run all tests coming from kubernetes fork, especially those introduced in patch releases;
+- start getting the correct single already during [kubernetes update process](https://github.com/openshift/kubernetes/blob/master/REBASE.openshift.md).
+
+### Tests co-located with code
+
+As of this writing (4.14 in progress), every major change in OpenShift requires
+the following steps:
+1. Modify [API](https://github.com/openshift/api/), [client-go](https://github.com/openshift/client-go/)
+   and optionally [library-go](https://github.com/openshift/library-go/).
+2. Modify the appropriate operator(s) repository(ies).
+3. Add new tests to [origin](https://github.com/openshift/origin/) repository.
+
+At any point in time during bumping one of the libraries in the repository mentioned
+in the last step above, one can expect to pick up additional, unrelated changes,
+which would require to either find the owner of that change, or react to those changes.
+
+Separating operator-specific tests to be co-located with the actual logic we will:
+- run the tests during submitting the actual change;
+- shorten the time when we start testing the new functionality;
+- limit the external factors which can affect the test merge.
+
+## Proposal
+
+### Workflow Description - old
+
+Currently, `openshift-tests run <test-suite>` does a few things:
+1. Discovers all available tests and splits them into suites based on test names/labels,
+   and test code paths (e.g. k8s package path).
+2. Filters tests based on: desired suite, configuration/environment of the cluster
+   being tested, parallel/serial requirements.
+3. Discovers the deployment topology and sets up all the arguments necessary
+   to monitor the cluster and run the tests.
+4. Starts monitoring the cluster functionality.
+5. Iterates over all the tests, creating a subprocess invoking `openshift-tests run-test`
+   passing a single test name to run as the argument, and reports back the result
+   to the main process.
+6. Present the summary results of running tests.
+
+Below diagram shows the current flow:
+
+![Old Flow](./old-flow.png)
+<!--
+Above diagram was generated at https://mermaid.live/ using following code:
+sequenceDiagram
+    actor U as User
+    participant O as openshift-tests
+    U->>+O: run openshift/conformance
+    O->>+O: list tests
+    loop over all tests
+        O->>+O: run #lt;test name#gt;
+        O->>+U: results
+    end
+-->
+
+### Workflow Description - new
+
+This proposal's goal is to extend test discovery (step 1), and running the actual
+test (step 5) with external binaries, which are retrieved from the payload image.
+The improved flow would be as follows:
+1. Retrieve test binaries from the payload image.
```

Comment:
this really means retrieve the set of test image(s) referenced by the payload image, and then retrieve the binaries from those images, right?  Since you seem to be assuming each binary goes in its own image (i'm still not sold on that, for the record :) )

Reply:
Good point - updated.

> (i'm still not sold on that, for the record :) )

it's an ongoing effort, I'll figure out a way to convince you :wink: